OpenLaTeXStudio/EditorModule/src/latexstudio/editor/EditorTopComponent.java

```diff
@@ -1,308 +1,322 @@
+    private void initComponents() {
+
+        jScrollPane1 = new javax.swing.JScrollPane();
+        rSyntaxTextArea = new org.fife.ui.rsyntaxtextarea.RSyntaxTextArea();
+
+        rSyntaxTextArea.setColumns(20);
+        rSyntaxTextArea.setRows(5);
+        rSyntaxTextArea.setSyntaxEditingStyle(org.openide.util.NbBundle.getMessage(EditorTopComponent.class, "EditorTopComponent.rSyntaxTextArea.syntaxEditingStyle")); // NOI18N
+        rSyntaxTextArea.addKeyListener(new java.awt.event.KeyAdapter() {
+            public void keyReleased(java.awt.event.KeyEvent evt) {
+                rSyntaxTextAreaKeyReleased(evt);
+            }
+            public void keyTyped(java.awt.event.KeyEvent evt) {
+                rSyntaxTextAreaKeyTyped(evt);
+            }
+        });
+        jScrollPane1.setViewportView(rSyntaxTextArea);
+
+        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(this);
+        this.setLayout(layout);
+        layout.setHorizontalGroup(
+            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+            .addGroup(layout.createSequentialGroup()
+                .addContainerGap()
+                .addComponent(jScrollPane1, javax.swing.GroupLayout.DEFAULT_SIZE, 380, Short.MAX_VALUE)
+                .addContainerGap())
+        );
+        layout.setVerticalGroup(
+            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+            .addGroup(layout.createSequentialGroup()
+                .addContainerGap()
+                .addComponent(jScrollPane1, javax.swing.GroupLayout.DEFAULT_SIZE, 278, Short.MAX_VALUE)
+                .addContainerGap())
+        );
+    }// </editor-fold>//GEN-END:initComponents
+
+    private void rSyntaxTextAreaKeyReleased(java.awt.event.KeyEvent evt) {//GEN-FIRST:event_rSyntaxTextAreaKeyReleased
+        dirty = true;
+    }//GEN-LAST:event_rSyntaxTextAreaKeyReleased
+
+    private void rSyntaxTextAreaKeyTyped(java.awt.event.KeyEvent evt) {//GEN-FIRST:event_rSyntaxTextAreaKeyTyped
+        if (currentFile == null || evt.isControlDown()) return;
+        setDisplayName(currentFile.getName() + '*');
+    }//GEN-LAST:event_rSyntaxTextAreaKeyTyped
+
+    // Variables declaration - do not modify//GEN-BEGIN:variables
+    private javax.swing.JScrollPane jScrollPane1;
+    private org.fife.ui.rsyntaxtextarea.RSyntaxTextArea rSyntaxTextArea;
+    // End of variables declaration//GEN-END:variables
+
+    @Override
+    public void componentOpened() {
+        ApplicationUtils.deleteTempFiles();
+        CompletionProvider provider = createCompletionProvider();
+        autoCompletion = new AutoCompletion(provider);
+        autoCompletion.setAutoActivationDelay(ApplicationSettings.INSTANCE.getAutoCompleteDelay());
+        autoCompletion.setAutoActivationEnabled(true);
+        autoCompletion.setAutoCompleteEnabled(true);
+        autoCompletion.install(rSyntaxTextArea);
+
```

Comment:
Somehow I don't get warm, fuzzy feelings from the fact that both in initialisation and listening we need to provide the mapping between ApplicationSettings and the current class.
But this is not really related to the current new setting, so I'm inclined to leave it like this while something better is thought of.